src/components/Authentication/LoginPage.js

```diff
@@ -38,20 +51,37 @@ const LOGIN = gql`
     }
 `;
 
+const GET_USER_TYPE = gql`
+    query GetUserType($username: String!){
+        userType(userName: $username)
+    }
+`
+
 const LoginPage = () => {
     const history = useHistory();
-    const {state} = useLocation();
+    const { state } = useLocation();
     const dispatch = useDispatch();
-    const token = useSelector(({auth}) => auth.token);
-
+    const token = useSelector(({ auth }) => auth.token);
+    const [userType, setUserType] = useState("");
     const [email, setEmail] = useState(state?.email);
+    const [realEmail, setRealEmail] = useState("");
     const [password, setPassword] = useState(null);
     const [shouldSave, setShouldSave] = useState(false);
     const [hasError, setHasError] = useState(false);
-    const [login, {loading}] = useMutation(LOGIN, {
+
+    const [getUserType, { data, loading }] = useLazyQuery(GET_USER_TYPE, {
+        variables: { "username": realEmail },
+        onCompleted: data => { setUserType(data.userType); 
+            if(userType===""){
+                setHasError(true);
```

Comment:
check if valid email here as well, then you don't need to have this on line 152 `error={hasError || email === ""}`